modules/placement-driver-api/src/testFixtures/java/org/apache/ignite/internal/placementdriver/TestPlacementDriver.java

```diff
@@ -37,13 +41,21 @@
 @TestOnly
 public class TestPlacementDriver extends AbstractEventProducer<PrimaryReplicaEvent, PrimaryReplicaEventParameters>
         implements PlacementDriver {
-    private final TestReplicaMetaImpl primaryReplica;
+    private final Supplier<TestReplicaMetaImpl> primaryReplicaSupplier;
 
     @Nullable
     private BiFunction<ReplicationGroupId, HybridTimestamp, CompletableFuture<ReplicaMeta>> awaitPrimaryReplicaFunction = null;
 
-    public TestPlacementDriver(String leaseholder) {
-        this.primaryReplica = new TestReplicaMetaImpl(leaseholder);
+    public TestPlacementDriver(ClusterNode leaseholder) {
```

Comment:
This is the test code and two additional constructors, this is just the sugar. It's no a problem.
Simply using only a pair of `nodeName` and `nodeId` will not work, see the reason here: [discussion_r1374357094](https://github.com/apache/ignite-3/pull/2741#discussion_r1374357094)
I will notice this in the javadoc for these constructors.

Reply:
Well, I can just repeat myself.

- We don't need ClusterService in TestPlacementDriver, we just need set of nodeName and nodeId for TestPrimaryReplica. It's too unclear if TestPlacementDriver will internally use service.localNode or similar.
- If you need some async logic, and probably you don't you can use supplier but again for nodeName and nodeId.
- We all understand that soon we will get rid of nodeName. And that nodeId will be the only param required. So, basically it's even no need to store nodeName just nodeId.

Reply:
> We don't need ClusterService in TestPlacementDriver, we just need set of nodeName and nodeId for TestPrimaryReplica. It's too unclear if TestPlacementDriver will internally use service.localNode or similar.

Needed, I gave examples above when we create `TestPlacementDriver`, and cannot get the `nodeId` until `ClusterService` has started.

Reply:
> Needed, I gave examples above when we create TestPlacementDriver, and cannot get the nodeId until ClusterService has started.

As it mentioned above, if nodeId is not known at the moment of creation, use supplier for it. You don't need a whole ClusterService. 

> And it will be very easy to change if we have implementations that allow us to simply throw out a couple of parameters in auxiliary constructors and methods.

You are right. So eventually we will have only:
`public TestPlacementDriver(Supplier<String> nodeIdSupplier)`
and
`public TestPlacementDriver(String nodeId)`

> For production code, I agree that it should be as you say, but we are working with a test implementation for which we can omit these formalities.

TestPlacementDriver should be as simple as possible. Having ClusterService there allows us to use even messaing service. Why do we need to propagate such a big variety of possibilities when we only need specific nodeId.
Besides that, common case for multiple nodes will be in using same primaryReplica for all of them. I mean following:
Let's assume that you have three nodes A, B and C. Each has it's own TestPlacementDriver instance however within each node we assign same as for primaryReplica, so having clusterService.topologyService().localMember() will be just incorrect.